mathesar/models.py

```diff
@@ -45,14 +49,23 @@ def get_queryset(self):
         return super().get_queryset()
 
 
-class DatabaseObject(BaseModel):
-    oid = models.IntegerField()
+class ReflectingObject(BaseModel):
     # The default manager, current_objects, does not reflect database objects.
     # This saves us from having to deal with Django trying to automatically reflect db
     # objects in the background when we might not expect it.
     current_objects = models.Manager()
     objects = DatabaseObjectManager()
 
+    class Meta:
+        abstract = True
+
+    def __str__(self):
+        return f"{self.__class__.__name__}"
+
+
+class DatabaseObject(ReflectingObject):
```

Comment:
I think `ReflectedObject` and `ReflectedObjectWithOID` would be clearer names here.